Druid/DRConstantConstraintSolver.class.st

```diff
@@ -20,8 +21,13 @@ DRConstantConstraintSolver >> collectConstraintsFrom: aDRValue [
 		ifPresentDo: [ ^ DRFullConstraint new ].
 
 	aDRValue isPhiFunction ifTrue: [
-		^ DRUnionConstraint constraints:
-			  (aDRValue operands collect: [ :op | self collectConstraintsFrom: op ]) ].
+		^ (phiReplacement isNil or: [ phiReplacement key ~= aDRValue ]) ifTrue: [
```

Comment:
Can we add a comment somewhere explaining how, when and why `phiReplacement` can be `nil`? :D